salt/cloud/__init__.py

```diff
@@ -1183,11 +1171,6 @@ def create(self, vm_, local_master=True):
             'minion', vm_, self.opts, default={}
         )
 
-        # Since using "provider: <provider-engine>" is deprecated, alias provider
-        # to use driver: "driver: <provider-engine>"
-        if 'provider' in vm_:
-            vm_['driver'] = vm_.pop('provider')
-
```

Comment:
@eradman Where are you setting the `rad_joyent_account:joyent` configuration? Perhaps I missed a use case when I originally made the deprecation, which of course would still be working if you hadn't moved over to `driver` yet.